src/geode/io/mesh/vtp_input.cpp

```diff
@@ -0,0 +1,424 @@
+            {
+                const auto nb_polygons =
+                    std::stoul( piece.attribute( "NumberOfPolys" ).value() );
+                if( nb_polygons == 0 )
+                {
+                    continue;
+                }
+                const auto nb_points =
+                    std::stoul( piece.attribute( "NumberOfPoints" ).value() );
+                // read_point_data( piece ); TODO for attributes
+                // read_cell_data( piece );
+                const auto points = read_points( piece, nb_points );
+                const auto polygons = read_polygons( piece, nb_polygons );
+
+                build_piece( points, polygons );
+            }
+        }
+
+        void build_piece( const absl::FixedArray< geode::Point3D >& points,
+            const absl::FixedArray< std::vector< geode::index_t > >&
+                polygon_vertices )
+        {
+            const auto nb_points = points.size();
+            const auto offset = surface_builder_->create_vertices( nb_points );
+            for( const auto p : geode::Range{ nb_points } )
+            {
+                surface_builder_->set_point( offset + p, points[p] );
+            }
+            absl::FixedArray< geode::index_t > new_polygons(
+                polygon_vertices.size() );
+            std::iota( new_polygons.begin(), new_polygons.end(),
+                surface_.nb_polygons() );
+            for( const auto& pv : polygon_vertices )
+            {
+                surface_builder_->create_polygon( pv );
+            }
+            surface_builder_->compute_polygon_adjacencies( new_polygons );
+        }
+
+        template < typename T >
+        std::vector< T > decode( absl::string_view input )
+        {
+            auto clean_input = remove_spaces( input );
+            auto fixed_header = clean_input.substr( 0, 16 );
+            std::string bytes;
+            auto decode_status = absl::Base64Unescape( fixed_header, &bytes );
+            OPENGEODE_EXCEPTION( decode_status,
+                "[VTPInput::decode] Error in decoding base64 "
+                "data for fixed header" );
+            OPENGEODE_ASSERT( bytes.size() == 12,
+                absl::StrCat( "[VTPInput::decode] Fixed header size is wrong "
+                              "(should be 12 bytes, got ",
+                    bytes.size(), " bytes)" ) );
+            const auto fixed_header_values =
+                reinterpret_cast< const geode::index_t* >(
+                    bytes.c_str() ); // should be unsigned long if UInt64
+            const auto nb_data_blocks = fixed_header_values[0];
+            if( nb_data_blocks == 0 )
+            {
+                return std::vector< T >{};
+            }
+            const auto uncompressed_block_size = fixed_header_values[1];
+            const auto nb_characters =
+                std::ceil( nb_data_blocks * 32. / 24. ) * 4;
+            auto optional_header = clean_input.substr( 16, nb_characters );
```

Comment:
What is 16? maybe sizeof(XXX)?

Reply:
Yes I add comments to explain all these magical numbers